### PR TITLE
[Enhancement] Implement `PartialEq` and `Eq` for `ExitCode`

### PR DESCRIPTION
**Pull request by kevinmatthes**
This PR adds the derived trait implementations of `PartialEq` and `Eq` for `ExitCode` as well as one corresponding and exhaustive unit test.

At the moment, I am working on a Rust binary crate returning `sysexits::ExitCode`s on exit.  The overall design of the code is that for every occurring error, an appropriate `sysexits::ExitCode` is propagated to the `main` function.  In order to define unit tests for the contained structs and traits, `sysexits::ExitCode`s need to be comparable for equality.

### CODE REVIEW

File: src/exit_code.rs [hunk 1]
```diff
@@ -17,7 +17,7 @@ use std::process::Termination;
 /// defined by [`<sysexits.h>`][sysexits-man-url].
 ///
 /// [sysexits-man-url]: https://man.openbsd.org/sysexits
-#[derive(Clone, Copy, Debug)]
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
```
Comment (sorairolake) [inline src/exit_code.rs:20]: Could you do this in alphabetical order?

**Review by sorairolake — CHANGES_REQUESTED** [review]
Thanks PR!

File: src/exit_code.rs (hunk 1 above)
Comment (kevinmatthes) [reply src/exit_code.rs]: Done.  :wrench: :heavy_check_mark:

I remember a restriction from an older version of Rust requiring this order as `PartialEq` is a requirement for `Eq`.  This was the reason why originally submitted it in the shown order.  It seems as if that restriction is removed, now.

**Review by sorairolake — APPROVED** [review]

**Comment by sorairolake** [comment]
@kevinmatthes Thanks!

**Pull request merged**